### PR TITLE
Expectation suite links and init flow update

### PR DESCRIPTION
**Pull request by roblim**
Addresses:
- https://app.asana.com/0/1119694864223484/1150577554651705/f
- https://app.asana.com/0/1119694864223484/1138929285394303/f

- DataDocs: Expectation Suite name on Validation Results page now links to appropriate page
- `great_expectations init`: cli now asks user if csv has header when adding a Spark Datasource with csv file

### CODE REVIEW

File: tests/cli/test_suite.py [hunk 1]
```diff
@@ -190,11 +190,14 @@ def test_suite_new_empty_suite_creates_empty_suite(
...
-        "comment": "New suite added via CLI",
-    }
+        'batch_kwargs': {
+            'datasource': 'mydatasource',
+            'path': csv,
+            'reader_method': 'read_csv'
+        },
+        'batch_markers': None,
+        'batch_parameters': None,
+        'comment': 'New suite added via CLI'}
```
Comment (Aylr) [inline tests/cli/test_suite.py:288]: I might be missing something here - I think this test uses a pandas datasource and it's not clear to me why the reader method is here.

File: great_expectations/cli/datasource.py [hunk 2]
```diff
@@ -1049,11 +1048,28 @@ def _get_batch_kwargs_from_generator_or_from_file_path(context, datasource_name,
...
             try:
+                batch_kwargs["reader_method"] = reader_method
+                if isinstance(datasource, SparkDFDatasource) and reader_method == "csv":
+                    header_row = click.confirm(
+                        "\nDoes this file contain a header row?",
+                        default=True
+                    )
+                    batch_kwargs["reader_options"] = {
+                        "header": header_row
+                    }
```
Comment (Aylr) [inline great_expectations/cli/datasource.py:1068]: Wonderful! I wonder how much effort would be involved in writing a test for this.

File: docs/changelog/changelog.rst [hunk 3]
```diff
@@ -2,7 +2,8 @@
 
 develop
 -----------------
-
+* DataDocs: Expectation Suite name on Validation Result pages now link to Expectation Suite page
```
Comment (Aylr) [inline docs/changelog/changelog.rst:5]: :heart:

**Review by Aylr — APPROVED** [review]
Looks great - I have a few questions that don't truly gate this.

File: tests/cli/test_suite.py (hunk 1 above)
Comment (roblim) [reply tests/cli/test_suite.py:288]: I assumed pandas datasource also takes reader_method since pandas_datasource.py has `guess_reader_method_from_path`

**Pull request merged**